app/src/main/java/swati4star/createpdf/util/MergeHelper.java

```diff
@@ -36,11 +43,57 @@ public MergeHelper(Activity activity, ViewFilesAdapter viewFilesAdapter) {
         mContext = mActivity;
         mViewFilesAdapter = viewFilesAdapter;
         mSharedPrefs = PreferenceManager.getDefaultSharedPreferences(mActivity);
+        mPDFEncryptUtils = new PDFEncryptionUtility(activity);
     }
 
     public void mergeFiles() {
         String[] pdfpaths = mViewFilesAdapter.getSelectedFilePath().toArray(new String[0]);
+        List<Integer> encryptedPfdPathsIndex = new ArrayList<>();
         String masterpwd = mSharedPrefs.getString(MASTER_PWD_STRING, appName);
+        int count = 0, i = 0;
+        for (String filePath : pdfpaths) {
+            if (mPDFEncryptUtils.isPDFEncrypted(filePath)) {
+                count++;
+                encryptedPfdPathsIndex.add(i);
+            }
+            i++;
+        }
+        MaterialDialog.Builder builder = new MaterialDialog.Builder(mActivity);
+        List<EditText> passwords = new ArrayList<>();
+        for (int k = 0; k < count; k++) {
+            TextView textView = new TextView(mActivity);
+            String[] fileName = pdfpaths[k].split(".\\/");
+            textView.setText(mContext.getString(R.string.enter_password_with_file_name)
+                    + fileName[fileName.length - 1]);
+            EditText pass = new EditText(mActivity);
+            pass.setHint(mContext.getString(R.string.enter_password_with_file_name) + fileName[fileName.length - 1]);
+            passwords.add(pass);
+            builder.customView(textView, true);
+            builder.customView(pass, true);
+        }
+        builder.positiveText("Merge");
```

Comment:
Here I was planning to keep a "merge" button for user to click after all the passwords are entered. But the problem is that for only one encrypted file the dialog box is appearing and rest of them are skipped.

Reply:
Instead of showing the dialogs at once after the PDFs are selected.

When user adds a PDF, if it is encrypted, we can ask them to enter password for that.

Reply:
yeah that can be done. But one issue is whenever the password is entered it creates a new decrypted file. What to do about that?

Reply:
should it be selected and unselect the encrypted file?

Reply:
yes. Select the unencrypted one & remove the encrypted.

Later on, we can delete the unecrypted one's.

Reply:
ok @Swati4star I'll try this today and update the pr. Thank you for the help :-)

Reply:
Great! Don't hesitate to ask if you get stuck a anywhere :)